plugins/prometheus/prometheus_test.go

```diff
@@ -42,10 +43,22 @@ func TestPromPlugin(t *testing.T) {
 			return nil
 
 		}
-		app.Run([]string{"", "-prometheus-port", "8080"})
+		app.Run([]string{"", "-prometheus-port", freeport()})
 	})
 }
 
+// freeport listens momentarily on port 0, then closes the connection and
```

Comment:
This approach seems like it would make for a flaky test, but I'm not sure what the alternative would be since you need to pass the port as an int to the app.

Reply:
Why would this flake?

Reply:
because something else could grab the port after you tested to see if it was available.  admittedly, not likely.

Reply:
Gotcha. I had a look around, and other people needing a known free port use this strategy too (eg github.com/phayes/freeport). I think I'd prefer to risk very unlikely flakes in return for testing this all the way through. The alternative would be to do something like keeping a port free on our testing machines, I guess.